ui/src/components/WorkerDetailsCard/index.jsx

```diff
@@ -1,52 +1,189 @@
-import React, { Component } from 'react';
+import React, { Component, Fragment } from 'react';
 import { format, parseISO } from 'date-fns';
-import List from '@material-ui/core/List';
-import ListItem from '@material-ui/core/ListItem';
-import ListItemText from '@material-ui/core/ListItemText';
+import {
+  Button,
+  List,
+  ListItem,
+  ListItemText,
+  Typography,
+} from '@material-ui/core';
+import DeleteIcon from 'mdi-react/DeleteIcon';
+import { WorkerManager } from 'taskcluster-client-web';
 import DateDistance from '../DateDistance';
+import DialogAction from '../DialogAction';
+import Label from '../Label';
+import { withAuth } from '../../utils/Auth';
 import { worker } from '../../utils/prop-types';
+import { NULL_PROVIDER } from '../../utils/constants';
 
 /**
  * Render information in a card layout about a worker.
  */
+@withAuth
 export default class WorkerDetailsCard extends Component {
   static propTypes = {
     /** A GraphQL worker response. */
     worker: worker.isRequired,
   };
 
+  state = {
+    dialogState: {
+      error: null,
+      open: false,
+      title: '',
+      body: '',
+      confirmText: '',
+      taskQueueId: '',
+      workerGroup: '',
+      workerId: '',
+    },
+  };
+
+  handleDialogActionOpen = (taskQueueId, workerGroup, workerId) => () => {
+    this.setState({
+      dialogState: {
+        open: true,
+        title: 'Terminate Worker?',
+        body: `This will terminate the worker with id ${workerId} in group ${workerGroup} within worker pool ${taskQueueId}.`,
+        confirmText: 'Terminate Worker',
+        taskQueueId,
+        workerGroup,
+        workerId,
+      },
+    });
+  };
+
+  handleDeleteClick = async () => {
+    const { taskQueueId, workerGroup, workerId } = this.state.dialogState;
+    const { user } = this.props;
+
+    this.setState({
+      dialogState: {
+        ...this.state.dialogState,
+        error: null,
+      },
+    });
+
+    try {
+      if (typeof user?.credentials === 'undefined') {
+        throw new Error('User credentials not found. Please log in.');
+      }
+
+      const wm = new WorkerManager({
```

Comment:
would be a good idea to move this logic to util function, like https://github.com/taskcluster/taskcluster/blob/3f2a65aa0945bcb3851eb0b4f5c27bfc19e695be/ui/src/utils/getArtifactUrl.js#L1-L12